File: code/queue/consumer.pl
use strict;
use warnings;
use Directory::Queue::Simple;

my $dirq = Directory::Queue::Simple->new(path => "/tmp/test");
my $done = 0;
for(my $name = $dirq->first(); $name; $name = $dirq->next()) {
    next unless $dirq->lock($name);
    my $data = $dirq->get($name);
    printf("Body: \"%s\"\n", $data);
    $dirq->remove($name);
    $done += 1;
}
printf("Consumed %s elements\n", $done);

File: code/queue/consumer.py
from dirq.QueueSimple import QueueSimple
from random import randint
import time

path = '/tmp/test'

print("start consuming...")
dirq = QueueSimple(path)
done = 0
while True:
    for name in dirq:
        # print("element: %s %s" % (path, name))
        if not dirq.lock(name):
            #print("couldn't lock: %s" % name)
            # name = dirq.next()
            continue
        element = dirq.get(name)
        #print(element.keys())
        print("Body: \"%s\"" % element)
        if randint(1,2) % 2:
            done += 1
            dirq.remove(name)
            #print('Removed')
        else:
            dirq.unlock(name)
        #name = dirq.next()
    print("consumed %i elements" % done)
    total_left = dirq.count()
    print("elements left in the queue: %d" % total_left)
    time.sleep(0.5)
